### PR TITLE
[PD-8997] UTM Parameter fix for AppApplyButton

### PR DESCRIPTION
**Pull request by nate998877**
## What does this PR do?
Fixes an issue where UTM parameters aren't properly applied to Apply urls. 

On first page load the `utm_` and `vs` where not present in the URL.

## Notes to Reviewer
You can use https://test-pd-8997--openresty-dejobs-org.microsites.recruitrooster.com/jobs to test

### CODE REVIEW

File: src/components/Jobs/AppApplyButton.vue [hunk 1]
```diff
@@ -49,8 +58,21 @@ export default {
                 return {}
             }
         },
+        setApplyUrl() {
+            if (this.mailto) {
+                return this.mailto
```
Comment (jeremytubbs) [inline src/components/Jobs/AppApplyButton.vue:63]: need to set `this.applyUrl = this.mailto` and return

File: src/components/Jobs/AppApplyButton.vue [hunk 2]
```diff
@@ -70,26 +92,11 @@ export default {
         },
         openApplyUrl() {
             if (process.client) {
+                this.setApplyUrl()
```
Comment (jeremytubbs) [inline src/components/Jobs/AppApplyButton.vue:95]: remove extra call to `setApplyUrl`

**Review by jeremytubbs — CHANGES_REQUESTED** [review]
fix tests with  `async () => {` and ` await wrapper.find('a').trigger('mouseenter')`

**Review by surgiie — APPROVED** [review]
Nothing in particular stands out to me that Jeremy hasnt already pointed out. Please address Jeremy's feedback.

File: src/components/Jobs/AppApplyButton.vue (hunk 2 above)
Comment (nate998877) [reply src/components/Jobs/AppApplyButton.vue:94]: The extra setApplyUrl is required as the initial hover won't always set the correct url depending on user behavior (loading page hovering on apply button & clicking, hovering quickly before sessionstorage is set). This appears to always set the correct url.

Comment (jeremytubbs) [reply src/components/Jobs/AppApplyButton.vue:94]: correct needed for firefox browser

**Review by jeremytubbs — APPROVED** [review]

**Pull request merged**